engine/rpcserver.go

```diff
@@ -3072,3 +3044,39 @@ func (s *RPCServer) GetRecentTrades(_ context.Context, r *gctrpc.GetSavedTradesR
 
 	return resp, nil
 }
+
+func (s *RPCServer) checkParams(exch string, a asset.Item, p currency.Pair) (exchange.IBotExchange, error) {
+	e := s.GetExchangeByName(exch)
+	if e == nil {
+		return nil, fmt.Errorf("exchange %s is not supported", exch)
+	}
+	if !e.IsEnabled() {
+		return nil, fmt.Errorf("exchange %s is not enabled", exch)
+	}
+	if a.IsValid() {
```

Comment:
What if its invalid?

Reply:
All the functions initially have a which is an assetType string. We then convert a to asset.Item using asset.New which returns an error. so that makes sure that the asset type is valid for the functions that require it to be. And then we input the asset.Item that we created into checkParams so in the case assetType is invalid that means that assetType must've been empty in the function that calls this checkParams

Reply:
Those checks occur outside this function and relies on each implementation being coded correctly. For a checking function, it would be better for the checking to occur within this to ensure that everything is set appropriately.
```
_, err := s.checkParams("Bitstamp", "", currency.Pair{})
```
returns no errors.

Also, please add test coverage

Reply:
I think this nit is overly pedantic and I don't think its worth worrying about the invalid scenario so much as its concerned with theoretical future bad implementations.